examples/host_guest/host_guest_hdr.py

```diff
@@ -0,0 +1,325 @@
+#! /usr/bin/env python
+"""
+Script for High Dynamic Range itc experiment.
+
+"""
+
+import numpy as np
+from itctools.itctools import ureg, Quantity
+from itctools.procedures import ITCProtocol, ITCExperimentSet, ITCExperiment, ITCHeuristicExperiment
+from itctools.materials import Solvent, Compound, SimpleSolution
+from itctools.labware import Labware, PipettingLocation
+
+
+
+#######################################
+# Below are the params needed to specify.
+#######################################
+
+
+# Define solvents.
+water = Solvent('water', density=0.9970479 * ureg.gram / ureg.milliliter)
+buffer = Solvent('buffer', density=1.014 * ureg.gram / ureg.milliliter)
+
+# The sample cell volume in microliters
+cell_volume = 202.8
+
+# Define compounds.
+
+# each experiment takes about 50 mins, cleaning takes 2.5
+nguests = 1
+assert(nguests==1)
+
+host = Compound('host', molecular_weight=1134.987 * ureg.gram / ureg.mole, purity=0.98)
+guest_molecular_weights = [123.11]
+
+
+# If HDR is not activated, there is only one value to guest_compound_Ka.
+# guest_compound_Ka = Quantity([np.exp(32)], ureg.liter / ureg.mole)  # see dGtoKa.py
+
+
+
+# c = [M]_0 * Ka
+# R_m = 6.4/c^0.2 + 13/c
+# Hence the scalling of c is eqivalent to the scalling of the association constant.
+
+guest_compound_Ka_guess = Quantity([np.exp(6), np.exp(8), np.exp(10), np.exp(12), np.exp(14)], ureg.liter / ureg.mole)
+
+# We only allow one guest to be added into the experiment, for now.
+guests = [
+    Compound(
+        name='guest%02d' %
+        (guest_index +
+         1),
+        molecular_weight=guest_molecular_weights[guest_index] *
+        ureg.gram / ureg.mole,
+        purity=0.975) for guest_index in range(nguests)]
+
+
+
+# Define troughs on the instrument.
+
+water_trough = Labware(RackLabel='Water', RackType='Trough 100ml')
+buffer_trough = Labware(RackLabel='Buffer', RackType='Trough 100ml')
+
+# Define source labware.
+source_plate = Labware(RackLabel='SourcePlate', RackType='5x3 Vial Holder')
+
+# Define source solutions on the deck.one
+# TODO : Use actual compound and solvent masses.
+# NOTE: Host solution is diluted by 10x.
+# NOTE: it might not be a good idea to further dilute.
+#       since the host solution is already very dilute because of the low solubility.
+
+
+host_solution = SimpleSolution(
+    compound=host,
+    compound_mass=11.54 *
+    ureg.milligram,
+    solvent=buffer,
+    solvent_mass=10.2002 *
+    ureg.gram,
+    location=PipettingLocation(
+        source_plate.RackLabel,
+        source_plate.RackType,
+        1))
+guest_solutions = list()
+
+# Dispensed by quantos
+guest_compound_masses = Quantity([1.38],
```

Comment:
We never want to dispense less than 10 mg to make a stock solution, since <10 mg quantities make it very difficult to accurately produce stock solutions.

Reply:
I'll increase the conc of the stock solution